File: backend/app/routes/sessions.py
from fastapi import APIRouter, Depends, status

from app.core.dependencies import get_current_user
from app.models.chat import MessageResponse
from app.models.session import CreateSessionRequest, SessionResponse, SessionListItemResponse, ReportResponse
from app.models.workflow import StartWorkflowResponse, WorkflowStatusResponse
from app.services import chat_service, session_service, workflow_service

router = APIRouter(prefix="/sessions", tags=["sessions"])


@router.post("", response_model=SessionResponse, status_code=status.HTTP_201_CREATED)
async def create_session(
    payload: CreateSessionRequest,
    user_id: str = Depends(get_current_user),
):
    return await session_service.create_session(payload, user_id)


@router.get("", response_model=list[SessionListItemResponse])
async def list_sessions(user_id: str = Depends(get_current_user)):
    return await session_service.get_sessions(user_id)


@router.get("/{session_id}", response_model=SessionResponse)
async def get_session(
    session_id: str,
    user_id: str = Depends(get_current_user),
):
    return await session_service.get_session(session_id, user_id)


@router.post("/{session_id}/run", response_model=StartWorkflowResponse, status_code=status.HTTP_202_ACCEPTED)
async def run_workflow(
    session_id: str,
    user_id: str = Depends(get_current_user),
):
    return await workflow_service.start_workflow(session_id, user_id)


@router.get("/{session_id}/status", response_model=WorkflowStatusResponse)
async def get_workflow_status(
    session_id: str,
    user_id: str = Depends(get_current_user),
):
    return await workflow_service.get_workflow_status(session_id, user_id)


@router.get("/{session_id}/report", response_model=ReportResponse)
async def get_report(
    session_id: str,
    user_id: str = Depends(get_current_user),
):
    return await session_service.get_report(session_id, user_id)


@router.get("/{session_id}/messages", response_model=list[MessageResponse])
async def get_messages(
    session_id: str,
    user_id: str = Depends(get_current_user),
):
    return await chat_service.get_messages(session_id, user_id)

File: frontend/src/api/sessions.ts
import { apiClient } from './client';
import type { Session, WorkflowStatus, ReportContent } from '../types';

export interface CreateSessionPayload {
  company_name: string;
  company_website: string;
  research_objective: string;
}

export interface WorkflowRunResponse {
  session_id: string;
  status: string;
  message: string;
}

export interface ReportResponse {
  session_id: string;
  generated_at: string;
  content: ReportContent;
}

export async function createSession(payload: CreateSessionPayload): Promise<Session> {
  const res = await apiClient.post<Session>('/sessions', payload);
  return res.data;
}

export async function getSessions(): Promise<Session[]> {
  const res = await apiClient.get<Session[]>('/sessions');
  return res.data;
}

export async function getSession(sessionId: string): Promise<Session> {
  const res = await apiClient.get<Session>(`/sessions/${sessionId}`);
  return res.data;
}

export async function runWorkflow(sessionId: string): Promise<WorkflowRunResponse> {
  const res = await apiClient.post<WorkflowRunResponse>(`/sessions/${sessionId}/run`);
  return res.data;
}

export async function getWorkflowStatus(sessionId: string): Promise<WorkflowStatus> {
  const res = await apiClient.get<WorkflowStatus>(`/sessions/${sessionId}/status`);
  return res.data;
}

export async function getReport(sessionId: string): Promise<ReportResponse> {
  const res = await apiClient.get<ReportResponse>(`/sessions/${sessionId}/report`);
  return res.data;
}
